corehq/util/markup.py

```diff
@@ -12,27 +12,29 @@
 )
 
 
+def _wrap_url(url):
+    return format_html('<a href="{url}">{url}</a>', url=url)
+
+
+def _get_url_segments(text):
+    for chunk in url_re.split(text):
+        # The match pattern contains multiple capture groups;
+        # Because only one will be populated, we need to ignore the Nones
+        if not chunk:
+            continue
+        elif url_re.match(chunk):
+            yield _wrap_url(chunk)
+        else:
+            yield escape(chunk)
+
+
 def mark_up_urls(text):
```

Comment:
(out of scope) I see this is only used by maintenance alerts. Is there any reason we don't just use markdown support for those? Seems like a one-line library call and a better overall experience for those reading the alerts + less code we have to maintain.

Reply:
That's probably true.  I agree that it's out of scope for the goal of this ticket, but maybe we can handle this in the future. Given that it's only used in one section of the code, however, I could also see the argument that its not worth fixing something that isn't broken. The only reason I touched this code was that [Bandit](https://bandit.readthedocs.io/en/latest/)'s static analysis would have complained about the `mark_safe` otherwise.